com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -72,12 +65,9 @@ public static int GetNumVisualInputs(Model model)
 
             foreach (var input in model.inputs)
             {
-                if (input.shape.Length == 4)
+                if (input.name.StartsWith(TensorNames.VisualObservationPlaceholderPrefix))
```

Comment:
Doesn't have to be in this PR, but you might want to consider moving these to the extension methods too.